src/CollectionTrait.php

```diff
@@ -34,6 +34,19 @@ public function add($value): self|static
         return $this;
     }
 
+    /** @return self[] */
+    public function chunk(int $size): array
+    {
+        if ($size < 1) {
+            return [$this];
+        }
+
+        return array_map(
+            static fn(array $chunk) => self::fromIterable($chunk),
+            array_chunk($this->toArray(), $size)
```

Comment:
You can not just **assume** that user `toArray` implementation will yield a valid array that can be used on the `fromIterable` method (which will call the `append` method for each member of the iterable).

You need to chunk the result of `$this->getArrayCopy()` instead, as that method will return an array with all the items in the collection, which should be valid "entries" for the `append` method.

```php
array_chunk($this->getArrayCopy(), $size)
```

Reply:
Thanks for the tip!